From 664634ca66bf069826734420f6324342bbb5e541 Mon Sep 17 00:00:00 2001
From: nickc <nickc@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 12 May 2011 09:44:14 +0000
Subject: [PATCH] 	* config/rx/rx.h (HAVE_PRE_DECREMENT): Fix typo.
 	* config/rx/rx.c (rx_is_legitimate_constant): Use gcc_unreachable
 	instead of abort.
 	(rx_align_for_label): Test label before extracting its number of
 	uses.
 	(rx_max_skip_for_label): Ignore debug insns.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@173694 138bc75d-0d04-0410-961f-82ee72b054a4

index 6fd1153..e0bd176 100644
--- a/gcc/config/rx/rx.c
+++ b/gcc/config/rx/rx.c
@@ -2451,8 +2451,7 @@ rx_is_legitimate_constant (rtx x)
 
 	default:
 	  /* FIXME: Can this ever happen ?  */
-	  abort ();
-	  return false;
+	  gcc_unreachable ();
 	}
       break;
       
@@ -2758,7 +2757,7 @@ rx_align_for_label (rtx lab, int uses_threshold)
      because the delay due to the inserted NOPs would be greater than the delay
      due to the misaligned branch.  If uses_threshold is zero then the alignment
      is always useful.  */
-  if (LABEL_NUSES (lab) < uses_threshold)
+  if (LABEL_P (lab) && LABEL_NUSES (lab) < uses_threshold)
     return 0;
 
   return optimize_size ? 1 : 3;
@@ -2775,7 +2774,7 @@ rx_max_skip_for_label (rtx lab)
   op = lab;
   do
     {
-      op = next_nonnote_insn (op);
+      op = next_nonnote_nondebug_insn (op);
     }
   while (op && (LABEL_P (op)
 		|| (INSN_P (op) && GET_CODE (PATTERN (op)) == USE)));
diff --git a/gcc/config/rx/rx.h b/gcc/config/rx/rx.h
index 4a88a5b..89f2e2c 100644
--- a/gcc/config/rx/rx.h
+++ b/gcc/config/rx/rx.h
@@ -155,7 +155,7 @@ extern enum rx_cpu_types  rx_cpu_type;
 
 #define LEGITIMATE_CONSTANT_P(X) 	rx_is_legitimate_constant (X)
 
-#define HAVE_PRE_DECCREMENT		1
+#define HAVE_PRE_DECREMENT		1
 #define HAVE_POST_INCREMENT		1
 
 #define MOVE_RATIO(SPEED) 		((SPEED) ? 4 : 2)
-- 
1.7.0.4

